### ISSUE TITLE
3-D Secure v2: Issue with Monzo in the UK

### ISSUE THREAD

**Issue by sandyscoffable** [LivotovLabs/3DSView#39]
### Steps to reproduce:
1. Submit a payment using a Monzo card that gets 3-D Secure v2 challenged
2. Break out of app containing 3DSView library to go to Monzo app and accept the payment
3. Return to original app

### What happens:
Term URL is hit, and `onAuthorizationCompleted3dsV2()` is not called with parameters (cres etc)

### What should happen:
`onAuthorizationCompleted3dsV2()` should be called, meaning the payment can be completed

**Comment by asifashraf270** [comment]
@sandyscoffable hi I am facing the same issue did you find the solution.?

**Comment by sandyscoffable** [comment]
I can't 100% remember. I think I managed to track down why in the end:

1. Webview is shown
2. User breaks out and goes to Monzo app
3. Android sees <app containing library> backgrounded and not being used, so pauses views etc after a short period of time (maybe less than a minute - varies, sometimes not enough time for user to authenticate with Monzo and authorise the payment)
4. The code I had to resurrect the app in the same state wasn't working, so when Android resumes the app, it's all messed up.

HTH.
